heron/tools/tracker/src/python/topology.py

```diff
@@ -589,30 +601,31 @@ def _update(
       scheduler_location=...,
     ) -> None:
     """Atomically update this instance to avoid inconsistent reads/writes from other threads."""
```

Comment:
If there was one topology per thread, then I don't think the lock would have an impact, but that's a big _if_ given I can't recall the threading structure. If that was the case, you'd probably want the lock for a consistent read, e.g. while doing a comparison/`__eq__`.

Reply:
???

Reply:
If the code below is,

When watching Zookeeper is used, Race Condition occurs at the same time because part [Link](https://github.com/apache/incubator-heron/pull/3830/files#diff-29e7366069fde10ea907fccddae5506e19a2d8dcaa5f515ce417fec4957e10ffR604-R631) is executed at the same time.

Reply:
```
heron update lad/www/beta heron-system-access-prometheus --component-parallelism parser:1
heron update lad/www/beta heron-system-access-prometheus --component-parallelism parser:2
```
Lock is required because the above ```_update``` is executed at the same time when instance update occurs.